src/pagopa.ts

```diff
@@ -374,3 +400,22 @@ export const rptIdFromPaymentNoticeQrCode = (
     organizationFiscalCode: paymentNoticeQrCode.organizationFiscalCode,
     paymentNoticeNumber: paymentNoticeQrCode.paymentNoticeNumber
   });
+
+/**
+ * Convert a QR code string into an RptId
+ * (the inverse function cannot exist because
+ * "amount" only exists in the QR code)
+ * @param qrCodeString string in the format PAGOPA|002|...
+ */
+export const rptIdFromQrCodeString = (
+  qrCodeString: string
+): Either<t.Errors, RptId> => {
+  const qrCode = PaymentNoticeQrCodeFromString.decode(qrCodeString);
```

Comment:
apart from naming (errorOrQrCode), as a shortcut you can just use:

`return PaymentNoticeQrCodeFromString.decode(qrCodeString).map(rptIdFromPaymentNoticeQrCode);`

Reply:
I'm afraid the version you proposed would not work, since `map` returns a `right(f(value))` and `f` (in this case, `rptIdFromPaymentNoticeQrCode`) returns an `Either` as well. 

An alternative I can think of uses `fold`, as follows:
```
PaymentNoticeQrCodeFromString.decode(qrCodeString).fold(left, rptIdFromPaymentNoticeQrCode)
```

Reply:
what about 

PaymentNoticeQrCodeFromString.decode(qrCodeString).chain(rptIdFromPaymentNoticeQrCode);